static/js/message_events.js

```diff
@@ -178,6 +178,10 @@ export function update_messages(events) {
         const $row = message_lists.current.get_row(event.message_id);
         if ($row.length > 0) {
             message_edit.end_message_row_edit($row);
+        } else {
+            // we should delete the message_id from currently_editing_messages
+            // if it exists there but we cannot find the row.
+            message_edit.remove_message_id_from_currently_editing_messages(event.message_id);
         }
```

Comment:
I think this is the right approach, but the wrong abstractions. I think the whole block of communicating with the `message_edit` module should be a single call to a new `message_edit.end_message_edit(message_id)`  function, and that function should be:
```
        const $row = message_lists.current.get_row(event.message_id);
        if ($row.length > 0) {
            message_edit.end_message_row_edit($row);
        } else if (currently_editing_messages.has(message_id)) {
            currently_editing_messages.delete(message_id);
        }
```

That provides a clear abstraction -- the `message_events` module is asking the `message_edit` module to mark the current message edit attempt as complete.

Reply:
Agreed, Thanks for this.
Updated the PR.